### DISCUSSION TITLE
Which Liquorix kernel version is valid for installation?

### DISCUSSION THREAD

**Discussion by Ricky-Tigg** [damentz/liquorix-package#160] [Q&A]
Good morning. As it stands, there is a _README.md_ file, however, there are no installation conditions mentioned in it. This is why the following aspect is unknown: Do the **full** version of the upstream project of **Liquorix kernel** and that of the Linux kernel in place of the installed Linux distribution have to be identical so that the installation of the Liquorix kernel can be operational?

**Comment by damentz** [comment]
> Good morning. As it stands, there is a _README.md_ file, however, there are no installation conditions mentioned in it. This is why the following aspect is unknown: Do the **full** version of the upstream project of **Liquorix kernel** and that of the Linux kernel in place of the installed Linux distribution have to be identical so that the installation of the Liquorix kernel can be operational?

I'm sorry, I really don't understand what you're asking.  Can you reword or clarify?

**Reply by Ricky-Tigg** [reply]
You will forgive me if I am on the wrong track while you fail to state what you said you do not understand; I would have to guess you, and I have little talent for that. So may what follows finally do your thing, otherwise it is pointless to continue.

- Upstream project version (_tag 6.7-7_) | "_update version to 6.7.3-3_".
- Linux kernel version (maintained by Linus Torvalds) installed on my Linux distribution:
```
$ uname -ro
6.7.3-200.fc39.x86_64 GNU/Linux
```
Thus as it appears, the versions here are identical – 6.7.3. And so, must it be like this?

Assumptions
Each release of this project supports the **ARM64** and **x86_64** architectures.

**Reply by damentz** [reply]
Ok sorry, I wasn't in a position to provide a full response.  Now that I am, let me try to figure out what you're asking:

> Do the full version of the upstream project of Liquorix kernel and that of the Linux kernel in place of the installed Linux distribution have to be identical so that the installation of the Liquorix kernel can be operational?

There is no _partial_ or _full_ version of Liquorix, there is just the version that is officially supported through the install script.  Kernel packages do not diff the linux image, it's a full compiled binary with its accompanying modules and scripts.  That is why I was confused as there's no concept of _full_, unless you're referring to a stripped down kernel that's deliberately smaller than the stock to fit on embedded systems.

Furthermore, because the package contents are distinct, there's no relationship or dependency on the kernel version on your system from the stock distro.  However, old userspace can be a huge issue, especially if firmware is ancient.  The older your distribution gets, the more likely unexpected behaviors with your system will occur, since new kernel versions are normally not tested against ancient userspace.

> Linux kernel version (maintained by Linus Torvalds) installed on my Linux distribution:

This is incorrect, this kernel is maintained by Red Hat / Fedora.  They add their own patches and opinionated config to get the intended behavior on Fedora.  If you encounter a bug with your kernel, you should report it to Fedora's bug tracker first and they will take care of determining whether it's a Fedora bug or upstream.

> Assumptions: Each release of this project supports the ARM64 and x86_64 architectures.

That's also incorrect, Liquorix does not support ARM64.  It _could_ work given you had a modified config and it built, but most ARM64 systems need out of tree patches and rarely work properly on mainline.  Liquorix would need a kernel for every popular baseboard out there to be properly usable, which as you can imagine, is a support and maintenance nightmare.

And last bit, you mention kernel versions.  Fedora uses a versioning scheme that indicates the upstream kernel version base plus the package version.  These two are distinct and tell you different information.  Liquorix's Debian version is the same; 6.7-7 means its the 7th iteration of the 6.7 package, and 6.7.3-3 means it's based on 6.7.3 and was the 3rd change to this stable update.

Now, there is no official Fedora kernel, but there have been some attempts on COPR with one package lasting a few years before becoming unmaintained.  I don't believe anyone has picked up the torch to continue, but I am a bit interested in providing an officially maintained kernel on COPR.  If I get some time to do research and experimentation I'll play around with it.

> You will forgive me if I am on the wrong track while you fail to state what you said you do not understand; I would have to guess you, and I have little talent for that. So may what follows finally do your thing, otherwise it is pointless to continue.

Checking your profile, you list Finland as your location.  I'm assuming English isn't your first language and the way you write seems unusual.  This may be a problem just with me, but FYI in-case you get similar feedback.

**Reply by Ricky-Tigg** [reply]
There is certainly no reason for you to be sorry, isn't it? Knowing whether everything pertinent that is worth mentioning in detail is present in this _README.md_ file is without doubt what matters if any. I have a profile indeed. I am not surprised that there must be ways for someone who feels the need to suggest, for example, that _`<this language>` is my first language_, to express it; here, **English**. But what do you expect me to do with your revelation? If only we knew what kind of reader you want to write for.

Return to the subject after this terminological digression on the version, which although having a certain educational interest, takes us away from the original subject. In my writings, the "_full version_" was to be interpreted only as the **version number of the release** (therefore as I invariably presented it – _6.7.3_) and this with regard to the **source of the Linux kernel**, on which the Linux distributions and the customized Liquorix kernel are built. Which interpretation was implicitly guided by my combination of expressions _Linux kernel version (maintained by Linus Torvalds) installed on my Linux distribution_ and _versions here are identical – 6.7.3_. However, everyone is obviously free to reformulate my writings if this can serve the understanding in any way of the participant in the discussion.

In short, should your writings be interpreted like this:
- Liquorix, in order to be installed, must first remove the upstream Linux kernel that is in place.
- Liquorix, when it comes to be built and installed, is independent of the upstream Linux kernel version that is place in the supported Linux distribution in which these operations take place.
Examples: Be _6.7.3_ for **x86_64** hardware the upstream Linux kernel version on which the kernel of a distribution supported by this project is built. Can we then install the Liquorix versions of linux-liquorix with these details: (syntax: _tag on this repository | upstream Linux kernel version merged to Liquorix_):
  - 6.7-8 | 6.7.4
  - 6.7-5 | 6.7.3
  - 6.6-17 | 6.6.13

**Comment by smxi** [comment]
I think you want to work on your communication skills, or if you are using something like ChatGPT to generate this text, stop doing that immediately because it's a really bad idea. I know scandinavians well, and they don't talk or think like this.

Nothing you said in your initial post, or even worse, in your response, had any real connection to technical reality, a quality I have seen only in fake ChatGPT/LLaMA type web pages that pretend to be about technical issues but are actually just random nonsense strung together.

Rather than respond with more of the same, pause, reflect, and ask yourself, why is my communication failing totally in this way? It's not about english as a first or second language, it's something else, but whatever that is, has nothing to do with the liquorix kernel.

**Reply by damentz** [reply]
Closing this thread, we suspect the OP to be an AI user and there's no objective for this post.
